--- a/include/LinkWebRTC.hpp
+++ b/include/LinkWebRTC.hpp
@@ -13,7 +13,7 @@
 #include <SDL.h>
 #endif
 
-#ifndef SWITCH
+#if !defined(SWITCH) && !defined(NO_ONLINE)
 #include "rtc/rtc.hpp"
 #include <future>
 #include <mutex>
@@ -38,7 +38,7 @@
 extern void initPlayer(u8 pId);
 #endif
 
-#ifndef SWITCH
+#if !defined(SWITCH) && !defined(NO_ONLINE)
 using nlohmann::json;
 using std::shared_ptr;
 using std::weak_ptr;
@@ -54,7 +54,7 @@
 };
 
 class LinkUniversal {
-#ifndef SWITCH
+#if !defined(SWITCH) && !defined(NO_ONLINE)
     static constexpr char STUN_SERVER[] = "stun:stun.l.google.com:19302";
     static constexpr char TYPE[] = "type";
     static constexpr char ID[] = "id";
